src/bootstrap/src/core/download.rs

```diff
@@ -748,3 +748,117 @@ fn path_is_dylib(path: &Path) -> bool {
     // The .so is not necessarily the extension, it might be libLLVM.so.18.1
     path.to_str().map_or(false, |path| path.contains(".so"))
 }
+
+/// Checks whether the CI rustc is available for the given target triple.
+pub(crate) fn is_ci_rustc_available_for_target(
+    target_triple: &str,
+    llvm_assertions: bool,
+) -> bool {
+    // FIXME: This list should be auto-generated.
+    const SUPPORTED_PLATFORMS: &[&str] = &[
+        "aarch64-apple-darwin",
+        "aarch64-apple-ios",
+        "aarch64-apple-ios-sim",
+        "aarch64-linux-android",
+        "aarch64-pc-windows-gnullvm",
+        "aarch64-pc-windows-msvc",
+        "aarch64-unknown-fuchsia",
```

Comment:
Are you sure this is right? The list I'd expect here is the [tier-2-with-host-tools list](https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-2-with-host-tools), since the tools are what we are downloading.

Reply:
I had a script that checked if the rust-std-nightly build was available for the target. This list was the result from it. I thought checking rust-std-nightly alone would be enough, but it turns out I should have checked for rustc-nightly and rustc-dev-nightly too. I updated the [script](https://gist.github.com/onur-ozkan/3a95499d5f681dcdd1f534d8f24c0de8) to check all three artifacts, and now the result matches the tier 1 + tier 2 list with host tools.